paimon-hive/paimon-hive-catalog/src/test/java/org/apache/paimon/hive/HiveCatalogTest.java

```diff
@@ -277,6 +279,47 @@ public void testAlterHiveTableParameters() {
         }
     }
 
+    @Test
+    public void testDropTable() {
+        try {
+            String databaseName = "test_db";
+            catalog.createDatabase(databaseName, false);
+            String tableName = "new_table";
+            Map<String, String> options = new HashMap<>();
+            Schema addHiveTableParametersSchema =
+                    new Schema(
+                            Lists.newArrayList(
+                                    new DataField(0, "pk", DataTypes.INT()),
+                                    new DataField(1, "col1", DataTypes.STRING()),
+                                    new DataField(2, "col2", DataTypes.STRING())),
+                            Collections.emptyList(),
+                            Collections.emptyList(),
+                            options,
+                            "");
+
+            catalog.createTable(
+                    Identifier.create(databaseName, tableName),
+                    addHiveTableParametersSchema,
+                    false);
+            // delete file path
+            HiveCatalog hiveCatalog = (HiveCatalog) catalog;
+            Path path = hiveCatalog.getTableLocation(Identifier.create(databaseName, tableName));
+            catalog.fileIO().deleteDirectoryQuietly(path);
+            assertThrows(
+                    Catalog.TableNotExistException.class,
+                    () -> catalog.getTable(Identifier.create(databaseName, tableName)));
+
+            List<String> tablesCon = catalog.listTables(databaseName);
+            assertThat(tablesCon).contains(tableName);
+
+            catalog.dropTable(Identifier.create(databaseName, tableName), false);
+            List<String> tables = catalog.listTables(databaseName);
+            assertThat(tableName).isNotIn(tables);
+        } catch (Exception e) {
```

Comment:
This try statement is very big, maybe add "throw Exception" in the method signature is better?